File: PLACEMENT_PREPARATION/ZOHO-2025/Gopinath/Day1/five.py
from typing import List
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        cur=0
        maxi=float('-inf')
        for i in range(len(nums)):
            cur+=nums[i]
            maxi=max(maxi,cur)
            if cur<0:
                cur=0
        return maxi

File: PLACEMENT_PREPARATION/Day1/Five.java
class Solution {
    public int maxSubArray(int[] nums) {
        int maxSum = nums[0];
        int currSum = nums[0];
        
        for (int i = 1; i < nums.length; i++) {
            currSum = Math.max(nums[i], currSum + nums[i]);
            maxSum = Math.max(maxSum, currSum);
        }
        
        return maxSum;
    }
}
